salt/salt.spec

```diff
@@ -791,10 +806,14 @@ install -Dd -m 0755 %{buildroot}%{_sysconfdir}/logrotate.d/
 install -Dpm 0644 salt/cli/support/profiles/* %{buildroot}%{python3_sitelib}/salt/cli/support/profiles
 
 # Install Salt tests
-install -Dd -m 0750 %{buildroot}%{_datadir}/salt
-install -Dd -m 0750 %{buildroot}%{_datadir}/salt/tests
-cp -a tests/* %{buildroot}%{_datadir}/salt/tests/
-sed -i '1s=^#!/usr/bin/\(python\|env python\)[0-9.]*=#!/usr/bin/python3=' %{buildroot}%{_datadir}/salt/tests/runtests.py
+install -Dd %{buildroot}%{python3_sitelib}/salt-testsuite
+cp -r -a tests %{buildroot}%{python3_sitelib}/salt-testsuite/
+# Remove runtests.py which is not used as deprecated method of running the tests
+rm %{buildroot}%{python3_sitelib}/salt-testsuite/tests/runtests.py
+# Copy conf files to the testsuite as they are used by the tests
+cp -r -a conf %{buildroot}%{python3_sitelib}/salt-testsuite/
+ln -s ../salt %{buildroot}%{python3_sitelib}/salt-testsuite/salt
```

Comment:
Why is this needed? Isn't it possible to import `salt` from `site-packages` without the symlink?

Reply:
> salt symlink is required for pytest to properly setup the testing environment in case of running all of the tests.

Why can't `pytest` import `salt` from `site-packages`?

> Some of the imorts are performed with test as module, so the name should be test but no salt-testsuite.

Understood.

Reply:
> Why can't `pytest` import `salt` from `site-packages`?
It's not about `pytest` in this case, it's related to `setup.py` trying to load something from `salt` placed nearby. I don't remember for sure now what exactly it was trying to load. Will check.

Reply:
Here I found the reason why I did it this way:
```
ImportError while importing test module '/usr/lib/python3.6/site-packages/salt-testsuite/tests/kitchen/test_kitchen.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/usr/lib64/python3.6/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
tests/kitchen/test_kitchen.py:10: in <module>
    import setup
E   ModuleNotFoundError: No module named 'setup'
```
and if put `setup.py` in place the following appears:
```
tests/kitchen/test_kitchen.py:10: in <module>
    import setup
setup.py:173: in <module>
    exec(compile(open(SALT_VERSION_MODULE).read(), SALT_VERSION_MODULE, "exec"))
E   FileNotFoundError: [Errno 2] No such file or directory: '/usr/lib/python3.6/site-packages/salt-testsuite/salt/version.py'
```
Maybe it's better to fix or just remove `kitchen` . Any suggestions are welcome. I just tried to keep everything untouched if possible.

Reply:
Hmmm, does it make any sense to run these `kitchen` tests at all? AFAIK this kitchen tool is just related with how upstream trigger tests execution, it is not testing any functionality provided by Salt.

I would vote to either fix this `test_kitchen.py` file or as Victor mentioned, just remove it from our Salt codebase (or `python3-salt-testsuite` package), so we avoid doing this hack with the `setup.py` in our specfile.

Reply:
link to `salt` and `setup.py` were deleted